src/scipp/compat/xarray_compat.py

```diff
@@ -30,71 +29,128 @@ def from_xarray(obj: Union[xr.DataArray, xr.Dataset]) -> VariableLike:
 
     See Also
     --------
-    scipp.compat.from_xarray_dataarray, scipp.compat.from_xarray_dataset
+    scipp.compat.to_xarray
     """
     import xarray as xr
-
-    if isinstance(obj, xr.DataArray):
-        return from_xarray_dataarray(obj)
+    if isinstance(obj, xr.Variable):
+        return _from_xarray_variable(obj)
+    elif isinstance(obj, xr.DataArray):
+        return _from_xarray_dataarray(obj)
     elif isinstance(obj, xr.Dataset):
-        return from_xarray_dataset(obj)
+        return _from_xarray_dataset(obj)
     else:
         raise ValueError(f"from_xarray: cannot convert type '{type(obj)}'")
 
 
-def _var_from_xarray(xr_obj: Union[xr.Coordinate, xr.DataArray]):
-    unit = xr_obj.attrs.get('units', '')
-    return Variable(dims=xr_obj.dims, values=xr_obj.values, unit=Unit(unit))
+def to_xarray(obj: VariableLike) -> Union[xr.Variable, xr.DataArray, xr.Dataset]:
+    """Convert a scipp object to the corresponding xarray object.
 
-
-def from_xarray_dataarray(da: xr.DataArray) -> DataArray:
-    """Converts an xarray.DataArray object to a scipp.DataArray object.
+    Warning
+    -------
+    Any masks and variances in the input will be stripped during the conversion.
 
     Parameters
     ----------
-    da:
-        An xarray.DataArray object to be converted.
+    obj:
+        The scipp object to convert.
 
     Returns
     -------
     :
-        The converted scipp DataArray object.
+        The converted xarray object.
 
     See Also
     --------
-    scipp.compat.from_xarray, scipp.compat.from_xarray_dataset
+    scipp.compat.from_xarray
+    """
+
+    if isinstance(obj, Variable):
+        return _to_xarray_variable(obj)
+    elif isinstance(obj, DataArray):
+        return _to_xarray_dataarray(obj)
+    elif isinstance(obj, Dataset):
+        return _to_xarray_dataset(obj)
+    else:
+        raise ValueError(f"to_xarray: cannot convert type '{type(obj)}'")
+
+
+def _from_xarray_variable(xr_obj: Union[xr.Coordinate, xr.DataArray]) -> Variable:
+    """Converts an xarray Coordinate or the data in a DataArray to a scipp.Variable.
+    """
+    unit = xr_obj.attrs.get('units', None)
+    return Variable(dims=xr_obj.dims,
+                    values=xr_obj.values,
+                    unit=Unit(unit) if unit is not None else default_unit)
+
+
+def _to_xarray_variable(var: Variable) -> xr.Variable:
+    """Converts a scipp.Variable to a dict containing dims, values and unit for storing
+    in either an xarray Coordinate or DataArray.
+    """
+    import xarray as xr
+    if var.bins is not None:
+        raise ValueError("Xarray does not support binned data.")
+    if var.variances is not None:
+        warn("Variances of variable were stripped when converting to Xarray.")
+    attrs = {'units': str(var.unit)} if var.unit is not None else None
+    return xr.Variable(dims=var.dims, data=var.values, attrs=attrs)
```

Comment:
This will break non-numpy types such as strings, but maybe that is fine for now?